charts/kubernetes-agent/templates/pod-rolebindings.yaml

```diff
@@ -19,6 +21,22 @@ roleRef:
   name: {{ $podClusterRoleName }}
```

Comment:
It might be better to switch here and modify the roleref - it makes it much easier to understand what's happening.

Reply:
Tried to make that fit - but couldn't so broke out the role-binding to another file

Reply:
We spoke about this, and decided that this was a much easier and more elegant way to do this. If anyone is curious, the following code should work if you did want to reuse the code:
```
{{- $targetNamespaces := list -}}
{{- if .Values.agent.deploymentTarget.enabled }}
{{- $targetNamespaces := list .Release.Namespace }}
{{- else }}
{{- $targetNamespaces := $.Values.scriptPods.serviceAccount.targetNamespaces }}
{{- end }}
{{- range $targetNamespace := $targetNamespaces }}
---
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  name: {{ printf "%s-binding" $podServiceAccountFullName }}
  namespace: {{ $targetNamespace | quote }}
subjects:
- kind: ServiceAccount
  name: {{ $podServiceAccountName }}
  namespace: {{ $.Release.Namespace | quote }}
roleRef:
{{- if .Values.agent.deploymentTarget.enabled }}
  kind: Role
  name: {{ $podWorkerRoleName }}
  apiGroup: rbac.authorization.k8s.io
{{- else }}
  kind: ClusterRole
  name: {{ $podClusterRoleName }}
  apiGroup: rbac.authorization.k8s.io
{{- end}}
{{- end -}}
```